packages/hierarchy-builder/src/hierarchy-builder/internal/operators/grouping/PropertiesGrouping.ts

```diff
@@ -0,0 +1,332 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { assert } from "@itwin/core-bentley";
+import { ProcessedInstanceHierarchyNode, PropertiesGroupingParams, PropertyGroup, PropertyGroupingNodeKey, Range } from "../../../HierarchyNode";
+import { ECClass, IMetadataProvider } from "../../../Metadata";
+import { createDefaultValueFormatter } from "../../../values/Formatting";
+import { TypedPrimitiveValue } from "../../../values/Values";
+import { getClass } from "../../Common";
+import { GroupingHandler, GroupingHandlerResult } from "../Grouping";
+
+interface PropertyInfo {
+  label: string;
+  propertyName: string;
+  fullClassName: string;
+  extraValues?:
+    | { formattedValue: string }
+    | {
+        fromValue: number;
+        toValue: number;
+      };
+}
```

Comment:
This structure is not really storing property info (only 2 attributes can be considered "property info"). The name needs improvement. 

Also, I'm not a fan of `extraValues` attribute. Maybe just store `PropertyGroupingNodeKey` here?

Reply:
Changed